ubuntu.json

```diff
@@ -143,6 +143,47 @@
       "ssh_wait_timeout": "10000s",
       "type": "parallels-iso",
       "vm_name": "{{ user `vm_name` }}"
+    },
+    {
+      "boot_command": [
+        "{{ user `boot_command_prefix` }}",
+        "/install/vmlinuz noapic ",
+        "file=/floppy/hyperv-{{ user `preseed` }} ",
+        "debian-installer={{ user `locale` }} auto locale={{ user `locale` }} kbd-chooser/method=us ",
+        "hostname={{ user `hostname` }} ",
+        "grub-installer/bootdev=/dev/sda<wait> ",
+        "fb=false debconf/frontend=noninteractive ",
+        "keyboard-configuration/modelcode=SKIP keyboard-configuration/layout=USA ",
+        "keyboard-configuration/variant=USA console-setup/ask_detect=false ",
+        "passwd/user-fullname={{ user `ssh_fullname` }} ",
+        "passwd/user-password={{ user `ssh_password` }} ",
+        "passwd/user-password-again={{ user `ssh_password` }} ",
+        "passwd/username={{ user `ssh_username` }} ",
+        "initrd=/install/initrd.gz -- <enter>"
```

Comment:
I'm trying your pull request with a cherry-pick to update to 16.04.3 ISO URL's (could you please rebase your PR?).

It seems that Packer has some problems with the boot_command. The last key strokes show some errors and I only get `initrd=/installngz`
<img width="1440" alt="bildschirmfoto 2017-08-29 um 11 32 36" src="https://user-images.githubusercontent.com/207759/29814566-12caf37e-8cae-11e7-92cf-ca6e140acc5d.png">

I had this several times. After manually fixing the string to `initrd=/install/initrd.gz --` and pressing ENTER the Packer build seems to work. I remember that other providers also had timing issues typing long boot commands.

My setup is an Azure VM running Windows Server 2016 with Packer 1.0.4.

One other thing to mention. We have to quote the file name for the `-var-file` option. Otherwise PowerShell ignores the `.json` extension. So my command to run your PR was:

```
 packer build -only=hyperv-iso -var-file="ubuntu1604.json" ubuntu.json
```

The Azure VM was created using a Terraform template: https://github.com/StefanScherer/packer-windows/tree/my/hyperv/terraform

<img width="1440" alt="bildschirmfoto 2017-08-29 um 11 44 06" src="https://user-images.githubusercontent.com/207759/29814972-68852f0e-8caf-11e7-849a-38e06a179b63.png">

Reply:
Hi Stefan, I have just merged my branch with master to upgrade to 16.04.3 ISO (no issue there). I can reproduce issues with `boot_command` you are mentioning on Azure VM as well. On physical machine this works as expected. 

I have quickly tested `"generation": "2"` VM where `boot_command` seems to be more reliable. Since floppy is not supported I had to use `preseed/url`. 

When testing, you will probably need to: 
 - run your own DHCP server such as http://www.dhcpserver.de/
 - enable firewall exceptions for packer to serve preseed via HTTP

Reply:
Thanks. In the meantime I've opened hashicorp/packer#5291 and it seems a bug in Packer. I went through the sources, but couldn't find the right place. It seems the hyperv builder uses some PowerShell code where the delay between each key stroke should be added.

My Azure VM has the DHCP feature installed and running, so this shouldn't be a problem. And yes, packer.exe should have a firewall exception as for other providers as well. 
I will try your latest commit tomorrow.